compiler/plugins/target/AMD-AIE/aie/AMDAIEObjectFifoStatefulTransform.cpp

```diff
@@ -176,9 +171,11 @@ class DMAChannelAnalysis {
       auto tile = memOp.getTile();
       for (auto &bl : r.getBlocks()) {
         for (auto op : bl.getOps<DMAStartOp>()) {
-          static_cast<DMAChannelDir>(op.getChannelDir()) == DMAChannelDir::MM2S
-              ? getProducerDMAChannel(tile)
-              : getConsumerDMAChannel(tile);
+          if (op.getChannelDir() == DMAChannelDir::MM2S) {
+            ++producerChannelsPerTile[tile];
```

Comment:
if you're going to do this, you might as well remove `getProducerDMAChannel` (and `getConsumerDMAChannel`).